pkg/datapath/loader/base.go

```diff
@@ -418,6 +392,27 @@ func (l *Loader) Reinitialize(ctx context.Context, o datapath.BaseProgramOwner,
 		log.Warning("Cannot check matching of C and Go common struct alignments due to old LLVM/clang version")
 	}
 
+	// Compile and load network programs, currently used for encryption.
+	if option.Config.EnableIPSec {
+		interfaces := option.Config.EncryptInterface
```

Comment:
I realized that I commented about converging this with the initial detection during daemon startup, but what happens if new ENI devices are added at runtime? Is the new version here properly detecting & applying the network progs to devices dynamically now?

Reply:
Can you add ENI devices at runtime without restarting the CNI? If so then we need to move it back.

Reply:
Yes, ENI devices get added dynamically. Main reason is that a single ENI only supports a few IP addresses (e.g. 8-ish) so new ENIs need to be created and attached as the number of pods on the node grows.

Reply:
OK so we need to move it back to its original location I guess :/